File: dclib/core/src/test/java/dclib/geometry/VectorUtilsTest.java
package dclib.geometry;

import com.badlogic.gdx.math.MathUtils;
import com.badlogic.gdx.math.Vector2;
import org.junit.Test;
import test.dclib.geometry.VectorTestUtils;

import static org.junit.Assert.assertEquals;

public final class VectorUtilsTest {

    @Test
    public void unit_ReturnsExpected() {
        Vector2 unit = VectorUtils.unit(new Vector2(2.5f, 2.5f));
        VectorTestUtils.assertEquals(0.70710678f, 0.70710678f, unit);
    }

    @Test
    public void offset_ReturnsExpected() {
        Vector2 vector1 = new Vector2(-4.3f, 2);
        Vector2 vector2 = new Vector2(1.6f, -6.7f);
        VectorTestUtils.assertEquals(5.9f, -8.7f, VectorUtils.offset(vector1, vector2));
    }

    @Test
    public void getLineX_ReturnsExpected() {
        Vector2 from = new Vector2(2, 4);
        Vector2 to = new Vector2(4, 8);
        float lineY = 6;
        assertEquals(3, VectorUtils.getLineX(from, to, lineY), MathUtils.FLOAT_ROUNDING_ERROR);
    }

    @Test
    public void fromAngle_VectorAndLength_ReturnsExpected() {
        Vector2 vector = VectorUtils.toVector2(45, 2.5f);
        assertEqualsVectors(vector, 1.76776695f, 1.76776695f);
    }

    private void assertEqualsVectors(final Vector2 vector, final float expectedX, final float expectedY) {
        assertEquals(expectedX, vector.x, MathUtils.FLOAT_ROUNDING_ERROR);
    }
}

File: dclib/core/src/test/java/dclib/geometry/VectorUtilsTest.kt
package dclib.geometry

import com.badlogic.gdx.math.MathUtils
import com.badlogic.gdx.math.Vector2
import org.junit.Assert.assertEquals
import org.junit.Test
import test.dclib.geometry.VectorTestUtils

class VectorUtilsTest {
    @Test
    fun unit_Base_Correct() {
        val unit = VectorUtils.unit(Vector2(2.5f, 2.5f))
        VectorTestUtils.assertEquals(0.70710678f, 0.70710678f, unit)
    }

    @Test
    fun offset_Base_Correct() {
        val vector1 = Vector2(-4.3f, 2f)
        val vector2 = Vector2(1.6f, -6.7f)
        VectorTestUtils.assertEquals(5.9f, -8.7f, VectorUtils.offset(vector1, vector2))
    }

    @Test
    fun getLineX_Base_Correct() {
        val from = Vector2(2f, 4f)
        val to = Vector2(4f, 8f)
        val lineY = 6f
        assertEquals(3f, VectorUtils.getLineX(from, to, lineY), MathUtils.FLOAT_ROUNDING_ERROR)
    }

    @Test
    fun toVector2_Base_Correct() {
        val vector = VectorUtils.toVector2(45f, 2.5f)
        VectorTestUtils.assertEquals(1.76776695f, 1.76776695f, vector)
    }
}
